File: Number_Theory_Functions/Factorial_array/fact.java
import java.util.*;

public class FactorialModulo {

    public static List<Long> factorial(int n, int p) {
        List<Long> res = new ArrayList<>(Collections.nCopies(n + 1, 1L));
        for (int i = 1; i <= n; i++) {
            res.set(i, (res.get(i - 1) * i) % p);
        }
        return res;
    }

    public static void main(String[] args) {
        int n = 25, p = 29;
        List<Long> res = factorial(n, p);

        for (Long i : res) {
            System.out.print(i + " ");
        }
    }
}

File: Number_Theory_Functions/Factorial_array/fact.cpp
#include <bits/stdc++.h>
using namespace std;

vector<long long> factorial(int n, int p) {
    vector<long long> res(n+1, 1);
    for(int i=1;i<=n;i++) {
        res[i] = (1LL * res[i-1] * i) % p;
    }

    return res;
}

int main() {
    int n = 25, p = 29;
    vector<long long> res = factorial(n, p);

    for(auto i:res) {
        cout<<i<<" ";
    }
}